pkgs/tools/archivers/unar/default.nix

```diff
@@ -0,0 +1,61 @@
+{ stdenv, fetchurl, gnustep, unzip, clang, bzip2 }:
+
+stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
+  pname = "unar";
+  version = "1.10.1";
+
+  src = fetchurl {
+    url = "http://unarchiver.c3.cx/downloads/${pname}${version}_src.zip";
+    sha256 = "0aq9zlar5vzr5qxphws8dm7ax60bsfsw77f4ciwa5dq5lla715j0";
+  };
+
+  buildInputs = [
+    gnustep.make unzip clang gnustep.base bzip2.dev
+  ];
+
+  postPatch = ''
+    substituteInPlace Makefile.linux \
+      --replace "CC = gcc" "CC=clang" \
+      --replace "CXX = g++" "CXX=clang++" \
+      --replace "OBJCC = gcc" "OBJCC=clang" \
+      --replace "OBJCXX = g++" "OBJCXX=clang++"
+
+    substituteInPlace ../UniversalDetector/Makefile.linux \
+      --replace "CC = gcc" "CC=clang" \
+      --replace "CXX = g++" "CXX=clang++" \
+      --replace "OBJCC = gcc" "OBJCC=clang" \
+      --replace "OBJCXX = g++" "OBJCXX=clang++"
+  '';
+
+  makefile = "Makefile.linux";
+
+  sourceRoot = "./The Unarchiver/XADMaster";
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp lsar $out/bin
+    cp unar $out/bin
+
+    mkdir -p $out/share/man/man1
+    cp ../Extra/lsar.1 $out/share/man/man1
+    cp ../Extra/unar.1 $out/share/man/man1
+
+    mkdir -p $out/etc/bash_completion.d
+    cp ../Extra/lsar.bash_completion $out/etc/bash_completion.d/lsar
+    cp ../Extra/unar.bash_completion $out/etc/bash_completion.d/unar
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = http://unarchiver.c3.cx/unarchiver;
+    description = "An archive unpacker program";
+    longDescription = ''
+      The Unarchiver is an archive unpacker program with support for the popular \
+      zip, RAR, 7z, tar, gzip, bzip2, LZMA, XZ, CAB, MSI, NSIS, EXE, ISO, BIN, \
+      and split file formats, as well as the old Stuffit, Stuffit X, DiskDouble, \
+      Compact Pro, Packit, cpio, compress (.Z), ARJ, ARC, PAK, ACE, ZOO, LZH, \
+      ADF, DMS, LZX, PowerPacker, LBR, Squeeze, Crunch, and other old formats. 
+    '';
+    license = with licenses; [ lgpl21Plus ];
+  };
```

Comment:
k. Though it should compile with gcc if we were to compile it with objc support using ```( gcc.cc.override { langObjC = true; } )``` in buildInputs. I just haven't tested it.